rpmlb/recipe.py

```diff
@@ -138,3 +252,13 @@ def make_sequence(name: str, count: int):
         sequence_map = dict(starmap(make_sequence, count_map.items()))
 
         return sequence_map
+
+
+class RecipeError(Exception):
+    """A class to manage validation error for recipe data."""
+
+    messages = {}
```

Comment:
This is not needed. Note that this is *class* attribute (belongs to the class, shared by all instances), whether `self.messages = messages` in `__init__` creates an *instance* attribute.

Effectively, you are masking this class attribute by instance-specific one, and the class one never actually get used. Just drop it :)